### DISCUSSION TITLE
Running multiple instances of AKHQ

### DISCUSSION THREAD

**Discussion by naveen8384** [tchiotludo/akhq#812] [Q&A] [answered]
Hi,
We are trying to running multiple instances of AKHQ in kubernetes but after logging in and when we click on any menu item it is redirecting to login page. 
It works well when we scale down AKHQ pods to 1.

Can someone help on running multiple akhq instance in Kubernetes?

**Comment by tchiotludo** [comment]
- What version of akhq are you using ? 
- Are you using a common [JWT secret](https://github.com/tchiotludo/akhq#jwt) on config or secrets ?

**Comment by naveen8384** [comment]
We are using akhq version 0.14.1 and secrets with basic-auth

**Reply by tchiotludo** [reply]
you will need to upgrade to latest version of akhq, previous version of akhq need to have a session stickyness on the load balancer to be able to scale because cookie are generated in memory. 
With latest version, the cookie depend on JWT secret that you can share between your instance with need a session stickyness.

**Reply by naveen8384** [reply]
Thanks @tchiotludo , I will try that.

**Reply by naveen8384** [reply]
I tried to update akhq version to 0.18.0. After deployment pods came up but akhq isn't loading. 
In logs I see this warning whenever I hit akhq url.

`2021-09-16 05:38:54,284 WARN  r-thread-1 org.akhq.log.access        [Date: 2021-09-16T05:38:54.28318Z] [Duration: 0 ms] [Url: GET /] [Status: 500] [Ip: /<ip>] [
User: -]`

I tried to update log level using below request even for that I didn't get any response

`curl -i -X POST -H "Content-Type: application/json" -d '{ "configuredLevel": "DEBUG" }'      http://localhost:8080/loggers/io.micronaut.configuration.security`

How can we check exact error and complete stack trace?

**Reply by tchiotludo** [reply]
Really space here. 
For the log, it's seems that you don't hit the endpoint (port 28081) 

If you want to have a lot of logs : 
```bash 
curl -i -X POST -H "Content-Type: application/json" -d '{ "configuredLevel": "TRACE" }' http://localhost:28081/loggers/root
```

**Reply by naveen8384** [reply]
It worked with old version after adding these annotations to service
    traefik.ingress.kubernetes.io/affinity: true,
    traefik.ingress.kubernetes.io/session-cookie-name: sticky
and with 0.18.0 version I updated micronaut security config and it worked after that

**Reply by ankurjains** [reply]
@naveen8384  I am using basic auth . How i configure same basic auth like below with jwt secret. in my applicaiton.yml there is no jwt secret config but only basic auth i have given iwth single pod. So I want to run multiple pod now. 

basic-auth:
      - username: admin1 # Username
        password: 8c6976e5b5410415bde908bd4dee15dfb167a9c873fc4bb8a81f6f2ab448a918
        passwordHash: SHA256
        groups: # Groups for the user
          - admin
      - username: appuser # Username
        password: 8c6976e5b5410415bde908bd4dee15dfb167a9c873fc4bb8a81f6f2ab448a918
        passwordHash: SHA256
        groups: # Groups for the user
          - topic-reader

**Reply by naveen8384** [reply]
To run multiple pods you have enable sticky sessions in service. If you are using traefik ingress then you need to add this two annotations in akhq values.yaml file

```
service:
  enabled: true
  type: ClusterIP
  port: 80
  annotations:
    traefik.ingress.kubernetes.io/affinity: "true"
    traefik.ingress.kubernetes.io/session-cookie-name: "sticky"
```

**Reply by ankurjains** [reply]
Yes 0.18.0 I used service without sticky in agic ingress controller  / nginx too and no need to of sticky now

**Comment by tchiotludo** [comment, marked as answer]
Use the last version, there is no need to have a sticky session, since we don't use cookie on server side but JWT token that is consistent between node.
